src/components/implementation/resmgr/naive/res_info.h

```diff
@@ -0,0 +1,132 @@
+#ifndef RES_INFO_H
+#define RES_INFO_H
+
+#include <consts.h>
+#include <cos_kernel_api.h>
+#include <cos_defkernel_api.h>
+#include <sl.h>
+#include <resmgr.h>
+#include <memmgr.h>
+
+#define RES_INFO_COMP_MAX_SUBTREE 8
+#define RES_INFO_COMP_MAX_THREADS (MAX_NUM_THREADS*RES_INFO_COMP_MAX_SUBTREE)
+
+extern u64_t res_info_schedbmp;
+
+struct res_shmem_glb_info {
+	int free_region_id;
+	int total_pages;
+
+	int npages[MEMMGR_MAX_SHMEM_REGIONS];
+};
+
+struct res_shmem_info {
+	struct cos_compinfo shcinfo;
+
+	vaddr_t shm_addr[MEMMGR_MAX_SHMEM_REGIONS];
+};
+
+struct res_comp_info {
+	spdid_t cid;
+	int thd_used;
+	struct cos_defcompinfo defci;
+	struct res_shmem_info shminfo;
+	struct sl_thd *tinfo[RES_INFO_COMP_MAX_THREADS]; /* including threads from components in subtree. */
```

Comment:
Though it makes sense for future extensions, I'm a little surprised to see `sl_thd`s here.  I don't see why we have the actual threads since we aren't scheduling them (yet, at least).